spec/lib/court_data_adaptor/query/defendant/by_name_spec.rb

```diff
@@ -26,53 +26,119 @@ def self.resource
+      end
+
+      it 'sends where query, unaltered, to resource' do
+        expect(resultset)
+          .to have_received(:where)
+          .with(name: 'GLove', date_of_birth: '1990-01-01')
+      end
+
+      it 'sends all message to resultset' do
+        expect(resultset).to have_received(:all)
+      end
     end
 
-    it 'populates prosecution_case_reference attribute' do
-      case_refs = results.map(&:prosecution_case_reference)
-      expect(case_refs).to be_present.and all(be_present)
+    context 'without mocking', :vcr do
+      it 'includes name filter in request' do
+        call
+        expect(
+          a_request(:get, %r{.*/api/internal/v1/prosecution_cases\?.*filter\[name\]=GLove.*})
+          .with(headers: { 'Content-Type' => 'application/vnd.api+json' })
+        ).to have_been_made.once
+      end
+
+      it 'includes date_of_birth filter in request' do
+        call
+        expect(
+          a_request(:get, %r{.*/api/internal/v1/prosecution_cases\?.*filter\[date_of_birth\]=1990-01-01.*})
+          .with(headers: { 'Content-Type' => 'application/vnd.api+json' })
+        ).to have_been_made.once
+      end
+
+      # NOTE: common platform appears to use elastic search
+      # so this matches names that include EITHER "trever" OR "glover"
+      context 'with two term name search', :vcr do
+        subject(:results) { instance.call }
+
+        let(:term) { 'trever glover' }
+
+        it 'returns defendant resources' do
+          expect(results).to all(be_instance_of(CourtDataAdaptor::Resource::Defendant))
+        end
+
+        it 'returns more than just exact match' do
+          expect(results.size).to be > 1
+        end
+
+        it 'returns exact name match first' do
+          expect(results.first).to have_attributes(name: 'Trever Glover', date_of_birth: '1990-01-01')
+        end
+
+        it 'returns defendants with exact matching date_of_birth' do
+          expect(results).to all(have_attributes(date_of_birth: '1990-01-01'))
+        end
+
+        it 'returns defendants with partial matching names' do
+          names = results.map { |res| res.name.downcase }
+          expect(names).to all(match('glover')).or all(match('trever'))
+        end
+
+        it 'populates prosecution_case_reference attribute' do
+          case_refs = results.map(&:prosecution_case_reference)
+          expect(case_refs).to be_present.and all(be_present)
+        end
+      end
+
+      # NOTE: common platform appears to use elastic search
+      # so this returns names that include "trever"
+      context 'with single term (partial) name search', :vcr do
+        subject(:results) { instance.call }
+
+        let(:term) { 'trever' }
+
+        it 'returns defendant resources' do
+          expect(results).to all(be_instance_of(CourtDataAdaptor::Resource::Defendant))
+        end
+
+        it 'returns more than just exact match' do
+          expect(results.size).to be 1
+        end
```

Comment:
Is this 'more than just exact match'?

With this and the one above lines 90-92 I think it would be clearer if something identifying the correct result(s) is shown rather than a count.

Reply:
Am getting rid of this check as is fairly meaningless, as name matching is handled in the spec at line 132

Reply:
thats done now